cpp/oneapi/dal/algo/kmeans_init/test/batch.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+
+namespace oneapi::dal::kmeans_init::test {
+
+namespace te = dal::test::engine;
+
+template <typename TestType>
+class kmeans_init_batch_test : public te::float_algo_fixture<std::tuple_element_t<0, TestType>> {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Method = std::tuple_element_t<1, TestType>;
+
+    auto get_descriptor(std::int64_t cluster_count) const {
+        return kmeans_init::descriptor<Float, Method>{ cluster_count };
+    }
+
+    bool not_available_on_device() {
+        constexpr bool is_plus_plus_dense =
+            std::is_same_v<Method, kmeans_init::method::plus_plus_dense>;
+        constexpr bool is_parallel_plus_dense =
+            std::is_same_v<Method, kmeans_init::method::parallel_plus_dense>;
+        return this->get_policy().is_gpu() && (is_plus_plus_dense || is_parallel_plus_dense);
+    }
+
+    constexpr bool is_dense() const {
+        return std::is_same_v<Method, method::dense> ||
+               std::is_same_v<Method, method::random_dense> ||
+               std::is_same_v<Method, method::plus_plus_dense> ||
+               std::is_same_v<Method, method::parallel_plus_dense>;
+    }
+
+    void dense_checks(std::int64_t cluster_count, const table& data) {
+        CAPTURE(cluster_count);
+
+        INFO("create descriptor")
+        const auto desc = get_descriptor(cluster_count);
+
+        INFO("compute");
+        const auto compute_results = compute(desc, data);
+        check_results(cluster_count, data, compute_results.get_centroids());
+    }
+
+    void check_results(std::int64_t cluster_count, const table& data, const table& centroids) {
+        std::int64_t row_count = data.get_row_count();
+        std::int64_t column_count = data.get_column_count();
+        ONEDAL_ASSERT(centroids.get_row_count() == cluster_count);
+        ONEDAL_ASSERT(centroids.get_column_count() == column_count);
+        const auto data_ptr = row_accessor<const float>(data).pull().get_data();
+        const auto centroid_ptr = row_accessor<const float>(centroids).pull().get_data();
+        std::int64_t match_count = 0;
+        std::vector<std::int64_t> indices;
+        for (std::int64_t i = 0; i < cluster_count; i++) {
+            for (std::int64_t j = 0; j < row_count; j++) {
+                bool match = true;
+                for (std::int64_t k = 0; k < column_count; k++) {
+                    if (data_ptr[j * column_count + k] != centroid_ptr[i * column_count + k]) {
+                        match = false;
+                        break;
+                    }
+                }
+                if (match) {
+                    if (std::count(std::begin(indices), std::end(indices), j) == 0) {
+                        indices.push_back(j);
+                        match_count++;
+                        break;
+                    }
```

Comment:
I guess, you could use `std::set`, `std::set::insert` and `std::set::count` instead. Or the same, but `std::unordered_set`

Reply:
Done